pkg/kubelet/server/stats/fs_resource_analyzer.go

```diff
@@ -89,7 +89,9 @@ func (s *fsResourceAnalyzer) updateCachedPodVolumeStats() {
 		podUid := pod.GetUID()
 		stats, found := s.getPodVolumeStats(pod)
 		if !found {
-			glog.Warningf("Could not locate volumes for pod %s", format.Pod(pod))
+			if pod.Status.Phase == api.PodRunning {
```

Comment:
Yeah. False positives are tolerable. We frequently get issues related to
stats collection failures. No logs might hinder debugging.

On Wed, Feb 17, 2016 at 1:51 PM, Phillip Wittrock <notifications@github.com>
wrote:

> In pkg/kubelet/server/stats/fs_resource_analyzer.go
> <https://github.com/kubernetes/kubernetes/pull/21355#discussion_r53237065>
> :
>
> > @@ -89,7 +89,9 @@ func (s *fsResourceAnalyzer) updateCachedPodVolumeStats() {
> >  		podUid := pod.GetUID()
> >  		stats, found := s.getPodVolumeStats(pod)
> >  		if !found {
> > -			glog.Warningf("Could not locate volumes for pod %s", format.Pod(pod))
> > +			if pod.Status.Phase == api.PodRunning {
>
> I am fine with that too. I think it is pretty low risk either way. The
> message should only be spammy of there are pods that are not setup for long
> periods of time.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/21355/files#r53237065>.
>